scala-generator/src/main/scala/models/Play2Json.scala

```diff
@@ -20,10 +20,11 @@ case class Play2JsonCommon(ssd: ScalaService) {
     }
   }
 
-  def implicitWriter(name: String, qualifiedName: String, methodName: String) = {
+  def implicitWriter(name: String, qualifiedName: String, methodName: String, isObject: Boolean) = {
+    val writerClass = if(isObject) "OWrites" else "Writes"
     Seq(
-      s"${implicitWriterDef(name)} = {",
-      s"  new play.api.libs.json.Writes[$qualifiedName] {",
+      s"${implicitWriterDef(name, writerClass)} = {",
+      s"  new play.api.libs.json.$writerClass[$qualifiedName] {",
       s"    def writes(obj: $qualifiedName) = {",
```

Comment:
We could also consider making the return type of this method explicit too (`JsObject` for `OWrites`, `JsValue` for everything else)